lib/commands/battery.js

```diff
@@ -1,14 +1,44 @@
+import log from '../logger.js';
+import { isIos18OrNewer } from '../utils.js';
+
 export default {
   /**
    * Reads the battery information from the device under test.
    *
    * This endpoint only returns reliable result on real devices.
    *
-   * @returns {Promise<BatteryInfo>} The battery info
+   * @returns {Promise<import('./types').BatteryInfo & {advance: Record<string, any>}>}
    * @this {import('../driver').XCUITestDriver}
    */
   async mobileGetBatteryInfo() {
-    return /** @type {BatteryInfo} */ (await this.proxyCommand('/wda/batteryInfo', 'GET'));
+    let batteryInfoFromShimService;
+    if (isIos18OrNewer(this.opts) && this.isRealDevice()) {
+      const {Services} = await import('appium-ios-remotexpc');
+      let remoteXPCConnection;
+      try {
+        let { diagnosticsService, remoteXPC } = await Services.startDiagnosticsService(this.device.udid);
+        remoteXPCConnection = remoteXPC;
```

Comment:
we may merge this expression with the above one. Check https://stackoverflow.com/questions/57065348/destructuring-and-rename-property